packages/server/lib/server-base.ts

```diff
@@ -183,13 +179,15 @@ export abstract class ServerBase<TSocket extends SocketE2E | SocketCt> {
     // This is handled separately from above since we may not need to delay the html
     // when we are in the scenario where the user creates the switchToDomain first
     // and then visits or navigates within the secondary domain
-    this.socket.localBus.on('ready:for:domain', (originPolicy) => {
+    this.socket.localBus.on('ready:for:domain', ({ originPolicy, failed }) => {
+      if (failed) return
```

Comment:
Sorry if I missed this, but do we have a test that verifies that this short circuits in the case `ready:for:domain` fails and we don't push a new remote state?

Reply:
There was kind of one in `response-middleware.spec.ts` but I went ahead and added some explicit unit tests. [cd8bc41](https://github.com/cypress-io/cypress/pull/20752/commits/cd8bc41b03a4057c18c137237e35abaa00eaf0c5)